File: src/main/java/core/procedures/characters/IntegerToChar.java
package core.procedures.characters;

import core.procedures.AFn;
import core.scm.FnArgs;

@FnArgs(minArgs = 1, maxArgs = 1, mandatoryArgsTypes = {Long.class})
public class IntegerToChar extends AFn {

  @Override
  public String getName() {
    return "integer->char";
  }

  @Override
  public Character apply1(Object arg) {
    return (char)((Number)arg).longValue();
  }
}

File: src/main/java/core/procedures/characters/IntegerToChar.kt
package core.procedures.characters

import core.procedures.AFn
import core.procedures.FnArgsBuilder

class IntegerToChar : AFn(FnArgsBuilder().min(1).max(1).mandatory(arrayOf<Class<*>>(Long::class.javaObjectType)).build()) {

    override val name: String
        get() = "integer->char"

    override fun apply1(arg: Any?): Char? {
        return (arg as Number).toLong().toChar()
    }
}